### PR TITLE
[DOC] Use run-aware CV in Haxby multiclass example

### LINKED ISSUE #6573: [DOC] Leakage in plot_haxby_multiclass feature selection
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe your proposed suggestion in detail.

This [paper](https://assets-eu.researchsquare.com/files/rs-10828698/v1_covered_8e877718-11d1-4548-a7d7-13c93cb85fc8.pdf?c=1787810853) raises a leakage issue in `examples/02_decoding/plot_haxby_multiclass.py` . Namely, the paper shows that the example uses feature selection directly in a 5 fold cv without taking into account run structure of the data, leading to increased performance. 

### List any pages that would be impacted.

_No response_

### PR DESCRIPTION
**Pull request by mohammedsadeghi**
- Closes #6573

Use of AI tools:
- [x] LLM tools were used to generate at least part of the content of this pull-request.
- [ ] No LLM tools were used to generate any part of the content of this pull-request.

Changes proposed in this pull request:

- use five-fold GroupKFold cross-validation with acquisition runs as groups in the Haxby multiclass example
- apply the same run-aware split to accuracy scoring and out-of-fold confusion-matrix predictions
- document the change in the changelog

Validation:

- ran the complete example with the non-interactive Matplotlib backend
- ran the repository hooks on the changed files

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
Since there are 12 runs, a 6-fold CV would make more sense IMHO.
LGTM otherwise.

File: examples/02_decoding/plot_haxby_multiclass.py [hunk 1]
```diff
@@ -112,19 +112,20 @@
 # --------------------------------------
 # The :term:`fMRI` data is acquired by runs,
 # and the noise is autocorrelated in a given run.
-# Hence, it is better to predict across runs when doing cross-validation.
-# Here we do a 5 fold cross-validation.
+# Hence, observations from the same run must not be split between the training
+# and test sets. Here we use a 5-fold group cross-validation, with runs as
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_multiclass.py]:
```suggestion
# and test sets. Here we use a 6-fold group cross-validation, with runs as
```

File: examples/02_decoding/plot_haxby_multiclass.py [hunk 2]
```diff
@@ -112,19 +112,20 @@
...
+# groups.
 
-from sklearn.model_selection import cross_val_score
+from sklearn.model_selection import GroupKFold, cross_val_score
 
-cv = 5
+cv = GroupKFold(n_splits=5)
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_multiclass.py]:
```suggestion
cv = GroupKFold(n_splits=6)
```

**Comment by Remi-Gau** [comment]
for the record

on main: 
image

in this PR: https://output.circle-artifacts.com/output/job/01b7cdd2-8c10-4f4b-a95f-67417e04c6fb/artifacts/0/dev/auto_examples/02_decoding/plot_haxby_multiclass.html#sphx-glr-auto-examples-02-decoding-plot-haxby-multiclass-py
image

**Review by Remi-Gau — APPROVED** [review]

**Review by pbarbarant — APPROVED** [review]
LGTM!

**Comment by Remi-Gau** [comment]
thanks @mohammedsadeghi
